src/cyw43_config.h

```diff
@@ -115,3 +115,31 @@
 #ifndef CYW43_EVENT_POLL_HOOK
 #define CYW43_EVENT_POLL_HOOK
 #endif
+
+#if LWIP_DHCP
+// If LWIP_DHCP is true, the default IP address has to be zero for isconnected()
+// to know when DHCP has assigned an IP address
+// If LWIP_DHCP is false, we expect CYW43_DEFAULT_IP_ADDRESS to be the set to
+// the static IP address
+#undef  CYW43_DEFAULT_IP_ADDRESS
+#endif
+
+#ifndef CYW43_DEFAULT_IP_ADDRESS
+#define CYW43_DEFAULT_IP_ADDRESS LWIP_MAKEU32(0, 0, 0, 0)
+#endif
+
+#ifndef CYW43_DEFAULT_IP_MASK
+#define CYW43_DEFAULT_IP_MASK LWIP_MAKEU32(255, 255, 255, 0)
+#endif
+
+#ifndef CYW43_DEFAULT_IP_GATEWAY
+#define CYW43_DEFAULT_IP_GATEWAY LWIP_MAKEU32(192, 168, 0, 1)
+#endif
+
+#ifndef CYW43_DEFAULT_IP_DNS
+#define CYW43_DEFAULT_IP_DNS LWIP_MAKEU32(8, 8, 8, 8)
+#endif
+
+#ifndef CYW43_DEFAULT_AP_IP_ADDRESS
+#define CYW43_DEFAULT_AP_IP_ADDRESS LWIP_MAKEU32(192, 168, 4, 1)
```

Comment:
Is it "correct" that `CYW43_DEFAULT_IP_GATEWAY` is in the `192.168.0.*` network, but `CYW43_DEFAULT_AP_IP_ADDRESS` is in the `192.168.4.*` network?

Reply:
I just copied the values that were in the code already. I'll have to check. It doesn't look entirely correct when the mask is 255.255.255.0

Reply:
It's correct as-is: there's a difference between STA and AP addresses.

In fact I think it would be best to rename the constants to the following, to make it clearer:
```c
CYW43_DEFAULT_IP_STA_ADDRESS
CYW43_DEFAULT_IP_STA_GATEWAY
CYW43_DEFAULT_IP_AP_ADDRESS
CYW43_DEFAULT_IP_AP_GATEWAY
```
(the last two will have the same value of 192.168.4.1 by default)

Reply:
Hopefully fixed as suggested. I left mask and dns alone as they're common between sta and ap.

Reply:
@peterharperuk there should be separate `AP_ADDRESS` and `AP_GATEWAY` (although default to the same value).

Also, did you want to name the STA specific ones `CYW43_DEFAULT_IP_STA_ADDRESS` and `CYW43_DEFAULT_IP_STA_GATEWAY`?

Reply:
added AP_ADDRESS and AP_GATEWAY and put STA in the other defines.